### PR TITLE
fix(ld-select): init ld-option-internal with up-to-date props

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

Fixes docs-pick-theme throwing exception on selection issue.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-select/ld-select.tsx [hunk 1]
```diff
@@ -363,17 +366,25 @@ export class LdSelect {
...
 
     if (!initialized) {
-      this.internalOptionsHTML = this.el.innerHTML['replaceAll'](
-        /<ld-option/g,
-        `<ld-option-internal${this.multiple ? ' mode="checkbox"' : ''}${
+      let internalOptionsHTML = ''
+      for (const ldOption of children) {
+        const classStr = ldOption.classList.length
+          ? ldOption.classList.toString()
+          : ''
```
Comment (renet) [inline src/liquid/components/ld-select/ld-select.tsx:373]:
```suggestion
        const classStr = ldOption.classList.toString()
```

Empty classList results in `''`.

File: src/liquid/components/ld-select/test/ld-select.spec.ts [hunk 2]
```diff
@@ -290,6 +290,48 @@ describe('ld-select', () => {
...
+    const { internalOptions } = await getInternalOptions(page)
+
+    const ldSelect = page.root
+
+    const spyInput = jest.fn()
+    let inputEvent
+    ldSelect.addEventListener('input', (ev) => {
+      inputEvent = ev
+      spyInput()
+    })
```
Comment (renet) [inline src/liquid/components/ld-select/test/ld-select.spec.ts:314]: This is a bit more complicated than it has to be. You could simply replace this with

```suggestion
    ldSelect.addEventListener('input', spyInput)
```

and later do something like

```js
expect(spyInput).toHaveBeenCalledWith(expect.objectContaining({
  detail: ["apple"],
})) // etc...
```

File: src/liquid/components/ld-select/ld-select.tsx [hunk 3]
```diff
@@ -363,17 +366,25 @@ export class LdSelect {
     }
 
     if (!initialized) {
-      this.internalOptionsHTML = this.el.innerHTML['replaceAll'](
-        /<ld-option/g,
-        `<ld-option-internal${this.multiple ? ' mode="checkbox"' : ''}${
+      let internalOptionsHTML = ''
```
Comment (renet) [inline src/liquid/components/ld-select/ld-select.tsx:369]: What is the difference in Output compared to the previous approach?

Comment (borisdiakur) [reply src/liquid/components/ld-select/ld-select.tsx:369]: The checked props were missing on the internal options with the previous solution. Now they are not missing.

Comment (renet) [reply src/liquid/components/ld-select/ld-select.tsx:369]: Shouldn't this result in a changed test somehow?

**Review by renet — CHANGES_REQUESTED** [review]
Maybe it's possible to (snapshot) test for the internal options using page.body ... ? If not, let's merge.

File: src/liquid/components/ld-select/ld-select.tsx (hunk 3 above)
Comment (borisdiakur) [reply src/liquid/components/ld-select/ld-select.tsx:369]: It doesn't. I can't figure out why this makes a difference for the docs-pick-theme component but no difference in the tests.

**Pull request merged**